### DISCUSSION TITLE
Missing Mandatory Constraint for BT-14-Lot in Notice Type 7

### DISCUSSION THREAD

**Discussion by dnl50** [OP-TED/eForms-SDK#747] [Q&A] [answered]
Hello there,

according to the `fields.json`, `BT-14-Lot` ist optional for Notice Type 7. But when no value is selected in `BT-14-Lot`, schematron rule `BR-BT-00615-0119` is violated. When I understand the underlying XPath expression correcty, `BT-14-Lot` should be mandatory in Notice Type 7.

Best Regards
Daniel

**Comment by YvesJo** [comment, marked as answer]
Hi,
The context for the rules is the node ND-LotProcurementDocument which may contain BT-14-Lot, BT-15-Lot, BT-615-Lot, BT-707-Lot, 
OPT-140-Lot.
BT-14-Lot is the "entry point" for the Procurement Documents, in its absence neither the listed fields, nor the parent node shall exist.
In absence of the parent, the rules wouldn't get fired.
Additionally, the existence of the parent node with the the absence of OPT-140-Lot would lead to a non XSD compliant XML instance, while OPT-140-Lot is only allowed when BT-14-Lot is defined.
The 42 condition on the conditionally mandatory rule for BT-615-Lot in Notice Subtype 7 is in fact useless.
Given these details, BT-14-Lot is Mandatory where Procurement Documents are mandatory and optional where these are optional and its existence and value are driving the existence of all the other Procurement Documents related fields.

**Reply by dnl50** [reply]
Hi there, thanks for the quick reply. Shouldn't `BT-14-Lot` always be marked as mandatory for all notice types then? Since `ND-LotProcurementDocument` is repeatable, one could just delete the `ND-LotProcurementDocument` node when no procurement documents are required. Or is this a problem since there is no _repeatable-mandatory_ and _repeatable-optional_ distinction in the fields.json?

**Reply by YvesJo** [reply]
Thanks for the suggestion. Making the field mandatory, would then force the parent node existence while it is optional for some notices. Displaying a mandatory field and keeping the possibility to delete the parent is a nice option and is currently limited to some specific cases with notices initially created from a previous notice (e.g. result notice where decision has been made for some of the lots only, contract modifications …).

**Reply by dnl50** [reply]
Does this mean that a repeatable is mandatory as long as it contains a _unconditionally mandatory_ BT?

**Reply by YvesJo** [reply]
Yes, for any Mandatory (without condition, i.e. stage 2 mandatory) field, for the field to exist, all of its ancestors have to exist.
